src/Components/AddAds/AddAds.tsx

```diff
@@ -0,0 +1,125 @@
+import { Button, FormControl, Modal } from '@mui/base';
+import { Grid, IconButton, Typography } from '@mui/material';
+import { Box } from '@mui/system';
+import axios from 'axios';
+import React, { useEffect, useState } from 'react'
+import { useForm } from 'react-hook-form';
+import CancelOutlinedIcon from '@mui/icons-material/CancelOutlined';
+import { useNavigate } from 'react-router';
+import { toast } from 'react-toastify';
+
+const style = {
+  position: 'absolute' as 'absolute',
+  top: '50%',
+  left: '50%',
+  transform: 'translate(-50%, -50%)',
+  width: 500,
+  bgcolor: '#203FC7',
+  borderRadius: "7px",
+  boxShadow: 24,
+  p: 5,
+};
+
+export default function AddAds() {
+  const { register, handleSubmit, formState: { errors } } = useForm();
+  const [rooms, setRooms] = useState([])
+  const [ads, setAds] = useState([])
+  let reqHeaders = 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI2NThhMTgyYjQ3ZWUyYjE0Zjk1NDY5OTAiLCJyb2xlIjoiYWRtaW4iLCJ2ZXJpZmllZCI6ZmFsc2UsImlhdCI6MTcwNDQ4NDEyNiwiZXhwIjoxNzA1NjkzNzI2fQ.N9gU4yHP3g8g5ajsm_Tf6w1EIDJE-Gfu4e0tsPejUj8'
+  let Headers = { Authorization: reqHeaders }
+  const navigate=useNavigate()
+
+console.log(errors);
+
+  {/*Get All Ads */ }
+  const getAllAds = () => {
+    axios.get('http://154.41.228.234:3000/api/v0/admin/ads', { headers: Headers }).then((response) => {
+      console.log(response.data.data.ads);
+      setAds(response.data.data.ads)
+    }).catch((error) => {
+      toast.error(error.response.message);
+    })
+
+
+  }
+
+
+  {/*create Ads */ }
+  const createAds = (data) => {
+    axios.post('http://154.41.228.234:3000/api/v0/admin/ads', data, { headers: Headers }).then((response) => {
+    toast.success("Added SuccessFully !")
+      navigate(-1)
+      getAllAds();
+
+    }).catch((error) => {
+   toast.error(error?.response?.data)
+
+    })
+  }
+  {/*Get All Rooms */ }
+
+  const getAllRooms = () => {
```

Comment:
You can remove the `getAllRooms` logic and replace it with the `useRooms` hook when it is merged  to the main.